src/vendedores.ts

```diff
@@ -1,31 +1,144 @@
-import { any } from "ramda";
+import R, { any } from "ramda";
 
 abstract class Vendedor {
+  certificacionesProductos?: number[] = [];
+  certificacionesSimples?: number[] = [];
+  firme?: boolean;
+  versatil?: boolean;
+  influyente!: boolean;
   abstract puedeTrabajarEn(ciudad: Ciudad): boolean;
 }
 
-export class VendedorFijo extends Vendedor {
-  constructor(public ciudadOrigen: Ciudad) {
+export class VendedorFijo extends Vendedor implements IVendedor, IInfluyente {
+  constructor(public ciudadOrigen: Ciudad,public certificacionesProductos: number[], public certificacionesSimples: number[]) {
     super();
+    // todo esto tendría que ir en un metodo de la clase abstracta pero no me saliógir
+    if (this.certificacionesProductos.length >= 1 && this.certificacionesProductos.length >= 1 && this.certificacionesProductos.length + this.certificacionesSimples.length >= 3 ){
+      this.versatil = true;
+    } else if (R.sum(this.certificacionesSimples) + R.sum(this.certificacionesProductos) >= 30) {
+      this.firme = true;
+    } else {
+      this.versatil = false;
+      this.firme = false;
+    }
+    // Defino Influyente como false
+    this.influyente = false;
+
+  }
+  esInfluyente(influyente: boolean): boolean {
+    return influyente == this.influyente;
+  }
+  
+  tieneCategoria(versatil: boolean, firme: boolean): boolean{
+    return(versatil == this.versatil, firme == this.firme);
   }
 
   puedeTrabajarEn(ciudad: Ciudad): boolean {
     return ciudad == this.ciudadOrigen;
   }
 }
 
-export class Viajante extends Vendedor {
-  constructor(public provinciasDondeTrabaja: Provincia[]) {
+export class Viajante extends Vendedor implements IVendedor, IInfluyente {
+  constructor(public provinciasDondeTrabaja: Provincia[]  ,public certificacionesProductos: number[], public certificacionesSimples: number[]) {
     super();
-  }
+    if (this.certificacionesProductos.length >= 1 && this.certificacionesProductos.length >= 1 && this.certificacionesProductos.length + this.certificacionesSimples.length >= 3 ){
+      this.versatil = true;
+    } else if (R.sum(this.certificacionesSimples) + R.sum(this.certificacionesProductos) >= 30) {
+      this.firme = true;
+    } else {
+      this.versatil = false;
+      this.firme = false;
+    }
+    // Esta es una solución muy pedorra para sumar las poblaciones. pero bue. Esto me tiraba error R.sum(this.provinciasDondeTrabaja[].poblacion) >= 10000000
+    
+    let poblacionTotal=0;
 
+    for (let i = 0; i < this.provinciasDondeTrabaja.length; i++) {
+      poblacionTotal += this.provinciasDondeTrabaja[i].poblacion
+    }
+    if (poblacionTotal >= 10000000){ //viajante.provinciasDondeTrabaja[0].poblacion
+      this.influyente = true;
+    } else {
+      this.influyente = false;
+    }
+
+  }
+  esInfluyente(influyente: boolean): boolean {
+    return influyente == this.influyente;
+  }
```

Comment:
Este método lo único que hace es comparar el booleano que vos le pasás con el que calculaste antes. Lo que se esperaba era que la lógica que resuelve la consulta estuviera acá. 

Algo así:

```suggestion
  esInfluyente(): boolean {
    return this.poblacionTotal() >= 10000000;
  }
```